packages/api-derive/src/tx/signingInfo.ts

```diff
@@ -31,28 +30,34 @@ function nextNonce (api: ApiInterfaceRx, address: string): Observable<Index> {
 }
 
 function signingHeader (api: ApiInterfaceRx): Observable<Header> {
-  return combineLatest([
-    api.rpc.chain.getHeader(),
-    api.rpc.chain.getFinalizedHead()
+  const finalized$ = api.rpc.chain.getFinalizedHead().pipe(
+    switchMap((finHash) => api.rpc.chain.getHeader(finHash)),
+    shareReplay(1)
```

Comment:
Not quite, because the resulting Observable that comes out from `combineLatest` won't emit until both inputs have emitted. Therefore, if the "current" observable returns a "genesis" header, we will still have to wait for:

```ts
api.rpc.chain.getFinalizedHead().pipe(
      switchMap((hash) =>
        api.rpc.chain.getHeader(hash)
      )
    )
```

to emit, when what it would be ideal is to cancel that request and emit/complete immediately.

Reply:
Well, we want both finalized as well as best to make a call - so, yes, both needs to be available.